apiclient/request_strategies.py

```diff
@@ -134,24 +134,20 @@ def _decode_response_data(self, response: Response):
 
     def _handle_bad_response(self, response: Response):
         """Convert the error into an understandable client exception."""
-        exception_class = self._get_exception_class(response.status_code)
+        exception_class = self._get_exception_from_client(response)
+        if not exception_class:
+            exception_class = self._get_exception_class(response.status_code)
         raise exception_class(
             message=f"{response.status_code} Error: {response.reason} for url: {response.url}",
             status_code=response.status_code,
             info=response.text,
         )
 
-    @staticmethod
-    def _get_exception_class(status_code: int) -> Type[exceptions.APIRequestError]:
-        if 300 <= status_code < 400:
-            exception_class = exceptions.RedirectionError
-        elif 400 <= status_code < 500:
-            exception_class = exceptions.ClientError
-        elif 500 <= status_code < 600:
-            exception_class = exceptions.ServerError
-        else:
-            exception_class = exceptions.UnexpectedError
-        return exception_class
+    def _get_exception_from_client(self, response):
+        return self.get_client().get_response_handler().get_exception_class(response)
```

Comment:
Remove this method.  In order to keep this simple, just using the exception handler on the parent client would be good enough for a customisable use case.

Reply:
👌